### PR TITLE
Merge 3.2.x

### PR DESCRIPTION
**Pull request by yakky**

### CODE REVIEW

File: cms/static/cms/js/modules/cms.plugins.js [hunk 1]
```diff
@@ -292,18 +292,20 @@ var CMS = window.CMS || {};
             }
 
             var bounds = this.options.plugin_restriction;
-            var type = clipboardPlugin.data('settings').plugin_type;
-            var parent_bounds = $.grep(clipboardPlugin.data('settings').plugin_parent_restriction, function (r) {
-                // special case when PlaceholderPlugin has a parent restriction named "0"
-                return r !== '0';
-            });
-            var currentPluginType = this.options.plugin_type;
+            if (clipboardPlugin.data('settings')) {
```
Comment (yakky) [inline cms/static/cms/js/modules/cms.plugins.js]: @vxsx could you check if this is the correct way to port #5097 FE code?

File: cms/static/cms/js/modules/cms.plugins.js [hunk 2]
```diff
@@ -292,18 +292,20 @@ var CMS = window.CMS || {};
...
-                pasteItem.addClass('cms-submenu-item-disabled');
-                pasteItem.find('.cms-submenu-item-paste-tooltip-restricted').css('display', 'block');
-                return false;
+                if ((bounds.length && $.inArray(type, bounds) === -1) ||
+                    (parent_bounds.length && $.inArray(currentPluginType, parent_bounds) === -1)) {
+                    pasteItem.addClass('cms-submenu-item-disabled');
+                    pasteItem.find('.cms-submenu-item-paste-tooltip-restricted').css('display', 'block');
+                    return false;
+                }
             }
```
Comment (vxsx) [inline cms/static/cms/js/modules/cms.plugins.js]: missing 
```
} else {
    return false;
}
```

Comment (yakky) [inline cms/static/cms/js/modules/cms.plugins.js]: Damn, thanks for spotting this
Updated

**Comment by vxsx** [comment]
:+1:

**Pull request merged**